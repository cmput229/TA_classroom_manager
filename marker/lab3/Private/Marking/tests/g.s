#-------------------------------
# Author: Kristen Newbury
# Date: August 1 2017
#
#
# test case for max functions and max function calls (10 each)
#
# lives at jal foo0: $s0
# lives at jal foo0: $t0
# lives at jal foo0:
# lives at jal foo0: $t2
# lives at jal foo0:
# lives at jal foo0:
# lives at jal foo0:
# lives at jal foo0:
# lives at jal foo0:
# lives at jal foo0: $t1
#
#-------------------------------
.text
main:

    jal     foo0
    add     $t2, $s0, $zero
    jr      $ra

foo0:
    jal     foo0
	addi	$t0, $t0, 13
	addi    $t2, $zero, 42
    jr      $ra
foo1:
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jal     foo0
    jr      $ra
foo2:
    addi	$t0, $zero, 13
    jal     foo0
    addi    $t2, $t2, 42
    jr      $ra
foo3:
    jal     foo0
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jr      $ra
foo4:
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jal     foo0
    jr      $ra
foo5:
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jal     foo0
    jr      $ra
foo6:
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jal     foo0
    jr      $ra
foo7:
    addi	$t0, $zero, 13
    addi    $t2, $zero, 42
    jal     foo0
    jr      $ra
foo8:
    jal     foo0
    addi	$t0, $zero, 13
    addi    $t2, $t1, 42
    jr      $ra
