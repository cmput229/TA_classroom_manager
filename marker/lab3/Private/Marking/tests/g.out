The live registers: $s0  
The live registers: $t0  
The live registers: 
The live registers: $t2  
The live registers: 
The live registers: 
The live registers: 
The live registers: 
The live registers: 
The live registers: $t1  
